backends/manager.go

```diff
@@ -0,0 +1,102 @@
+package backends
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/theupdateframework/notary/tuf/utils"
+)
+
+// Backend info for available backends
+type Backend struct {
+	Name           string   `json:",omitempty"`
+	Path           string   `json:"-"`
+	Version        string   `json:",omitempty"`
+	SupportedTypes []string `json:"-"`
+	Err            error    `json:"-"`
+}
+
+// BackendMetadata backend metadata
+type BackendMetadata struct {
+	Name    string `json:",omitempty"`
+	Version string `json:",omitempty"`
+}
+
+const (
+	backendPrefix = "docker-"
+	backendSuffix = "-backend"
+)
+
+var allowedBackends = map[string][]string{"docker-compose-cli-backend": {"aci", "ecs", "local"}}
+
+// ListBackends produces a list of available backends on the system and their context types
+func ListBackends() []Backend {
+	return listBackendsFrom(getDockerCliBackendDir(), getMetadata, allowedBackends)
```

Comment:
question: Shouldn't this function return an error? As it looks like there are many cases when `listBackendsFrom` returns an empty slice, for example if `os.Stat` on the backend directory returns an error. This error is ignored and maybe it could help the user to understand why we can't find any backend?

Reply:
Right, I returned an error and log the error meg to debug if there is one, that will help debuging when something goes wrong